src/main/kotlin/lotto/model/LottoTicket.kt

```diff
@@ -1,16 +1,29 @@
 package lotto.model
 
-data class LottoTicket(val numbers: LottoNumbers = LottoNumbers.autoCreate()) {
+import lotto.model.number.LottoNumber
+import lotto.model.number.LottoNumbers
+import lotto.model.number.LottoNumbersFactory
+import lotto.model.number.WinningNumber
+import lotto.model.number.WinningNumbers
 
-    constructor(numbers: List<Int>) : this(LottoNumbers(numbers))
+class LottoTicket(val candidateNumbers: LottoNumbers) {
+    constructor(candidateNumbers: List<Int>) : this(LottoNumbersFactory.create(candidateNumbers))
 
-    fun countMatch(targetNumbers: Set<LottoNumber>): Int {
-        return targetNumbers.count {
-            numbers.contains(it)
-        }
+    constructor(generator: NumbersGenerator = RandomNumbersGenerator()) : this(
+        LottoNumbersFactory.create(generator.getNumbers(LottoNumber.MAXIMUM, LottoNumbers.CANDIDATE_SIZE))
+    )
+
+    fun isMatch(targetNumber: WinningNumber): Boolean {
+        return candidateNumbers.contains(targetNumber)
+    }
+
+    fun countMatch(targetNumbers: WinningNumbers): Int {
```

Comment:
`WinningNumbers` 대신 `LottoNumbers`를 사용하면 어떨까요?

```suggestion
    fun countMatch(targetNumbers: LottoNumbers): Int {
```